### PR TITLE
fix: improve ziti identity handling

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add reauthentication for ziti management client calls on 401 responses
- resolve managed identities by identity id or agent-prefixed id
- add store lookup for managed identities by identity id

## Testing
- go vet ./...
- go test ./...

#1

### CODE REVIEW

**Comment by casey-brooks** [comment]
## Summary
- add reauthentication around ziti management client calls on 401 responses
- resolve managed identities by UUID or agent-prefixed identity ids
- add store lookup for managed identities by identity id

## Testing
- go vet ./...
- go test ./...

**Comment by casey-brooks** [comment]
## Summary
- add reauthentication around ziti management client calls on 401 responses
- resolve managed identities by UUID or agent-prefixed identity ids
- add store lookup for managed identities by identity id

## Testing & Lint
- go vet ./... (no issues)
- go test ./... (passed: 2, failed: 0, skipped: 0)

File: internal/ziti/client.go [hunk 1]
```diff
@@ -83,6 +107,50 @@ func loadCAPool(caFile string) (*x509.CertPool, error) {
...
+}
+
+func isUnauthorized(err error) bool {
+	if err == nil {
+		return false
+	}
+	var checker statusCodeChecker
+	if errors.As(err, &checker) && checker.IsCode(401) {
+		return true
+	}
```
Comment (noa-lucent) [inline internal/ziti/client.go:149]: [major] The `strings.Contains(err.Error(), "[401]")` fallback is fragile. It matches any error message that happens to contain the substring `[401]`, including unrelated log messages or wrapped errors. This is a classic AI-generated "just-in-case" fallback that undermines the structured `IsCode` check above it.

The `go-openapi` generated error types from `edge-api` implement `IsCode()`. If you've encountered a case where `errors.As` doesn't match, the correct fix is to identify that specific error type and add it to the interface check — not to fall back to string matching.

Remove the `strings.Contains` branch. If a concrete edge case exists where `IsCode` is insufficient, document it and handle that specific type.

File: internal/server/server.go [hunk 2]
```diff
@@ -258,6 +258,28 @@ func (s *Server) ResolveIdentity(ctx context.Context, req *zitimanagementv1.Reso
...
 }
 
+func (s *Server) resolveManagedIdentity(ctx context.Context, zitiID string) (store.ManagedIdentity, error) {
+	if identityID, ok := parseManagedIdentityID(zitiID); ok {
+		return s.store.ResolveIdentityByIdentityID(ctx, identityID)
+	}
+	return s.store.ResolveIdentity(ctx, zitiID)
+}
+
+func parseManagedIdentityID(value string) (uuid.UUID, bool) {
```
Comment (noa-lucent) [inline internal/server/server.go:268]: [minor] `parseManagedIdentityID` has no unit tests. It has non-trivial parsing logic (raw UUID, `agent-` prefixed UUID extraction) that should be covered — especially edge cases like `agent-` followed by an invalid UUID, or a value that is neither a UUID nor agent-prefixed.

File: internal/store/store.go [hunk 3]
```diff
@@ -55,6 +56,18 @@ func (s *Store) ResolveIdentity(ctx context.Context, zitiIdentityID string) (Man
...
 
+func (s *Store) ResolveIdentityByIdentityID(ctx context.Context, identityID uuid.UUID) (ManagedIdentity, error) {
+	row := s.pool.QueryRow(ctx, `SELECT ziti_identity_id, identity_type, ziti_service_id, created_at FROM managed_identities WHERE identity_id = $1`, identityID)
+	identity := ManagedIdentity{IdentityID: identityID}
+	if err := row.Scan(&identity.ZitiIdentityID, &identity.IdentityType, &identity.ZitiServiceID, &identity.CreatedAt); err != nil {
+		if errors.Is(err, pgx.ErrNoRows) {
+			return ManagedIdentity{}, ErrManagedIdentityNotFound
+		}
+		return ManagedIdentity{}, fmt.Errorf("resolve identity by identity id: %w", err)
+	}
```
Comment (noa-lucent) [inline internal/store/store.go:67]: [minor] `ResolveIdentityByIdentityID` queries `WHERE identity_id = $1`, but there is no index on the `identity_id` column (checked all migrations). This will result in a sequential scan. Since `ResolveIdentity` is documented as a hot path (used by Gateway), add an index:

```sql
CREATE INDEX idx_managed_identities_identity_id ON managed_identities (identity_id);
```

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Overall this PR is well-structured. The re-authentication pattern via `withReauth` is clean, the interface extraction for `identityService`/`serviceService` enables good testability, and the enrollment flow in `enroll.go` is thorough with solid test coverage.

**Blocking:**
- The `strings.Contains(err.Error(), "[401]")` fallback in `isUnauthorized` is fragile and must be removed in favor of structured error checking only.

**Non-blocking but recommended:**
- Add unit tests for `parseManagedIdentityID`.
- Add a migration for an index on `identity_id` to support the new `ResolveIdentityByIdentityID` query path.

File: internal/ziti/client.go (hunk 1 above)
Comment (casey-brooks) [reply internal/ziti/client.go:149]: Removed the string-matching fallback; `isUnauthorized` now relies solely on `IsCode(401)`.

**Comment by casey-brooks** [comment]
## Summary
- remove `isUnauthorized` string-matching fallback and rely on structured `IsCode(401)` checks
- add `TestParseManagedIdentityID` coverage for UUID/agent prefix/invalid cases
- add migration `0005_add_identity_id_index.sql` indexing `managed_identities.identity_id`

## Testing & Lint
- go vet ./... (no issues)
- go test ./... (passed: 2, failed: 0, skipped: 0)

**Review by noa-lucent — APPROVED** [review]
All 3 review items verified in code:

1. **`isUnauthorized` string fallback removed** — now returns `false` instead of `strings.Contains`, unused `"strings"` import cleaned up.
2. **`parseManagedIdentityID` tests added** — covers raw UUID, agent-prefixed UUID, invalid UUID, short prefix, and empty string.
3. **`identity_id` index added** — new migration `0005_add_identity_id_index.sql`.

Clean commit, no new issues. Approving.
